cmd/ddev/cmd/commands.go

```diff
@@ -208,6 +211,20 @@ func addCustomCommands(rootCmd *cobra.Command) error {
 					commandToAdd.Run = makeContainerCmd(app, inContainerFullPath, commandName, service, execRaw, relative)
 				}
 
+				if disableFlags {
+					// Hide -h because we are disabling flags
+					// Also hide --json-output for the same reason
+					// @see https://github.com/spf13/cobra/issues/1328
+					commandToAdd.InitDefaultHelpFlag()
+					err = commandToAdd.Flags().MarkHidden("help")
+					originalHelpFunc := commandToAdd.HelpFunc()
+					if err == nil {
+						commandToAdd.SetHelpFunc(func(command *cobra.Command, strings []string) {
+							_ = command.Flags().MarkHidden("json-output")
```

Comment:
Wondering if `--json-output` does make sense for custom commands at all, maybe we should hide it always for custom commands. What do you think @rfay ?

Reply:
I think this is a good idea.

Reply:
Sounds reasonable to me. Not sure how nonfunctional json output could be useful.

Reply:
I just tried it and some things kind of work:

```
ddev platform --json-output list

                                              
  [RuntimeException]                          
  The "--json-output" option does not exist.  
                                              

Usage: list [--raw] [--format FORMAT] [--all] [--] [<namespace>]

For more information, type: platform help list

{"level":"fatal","msg":"\u001b[31mFailed to run platform --json-output list: exit status 1\u001b[0m","time":"2022-10-13T09:00:53-03:00"}
```

In the example above, `--json-output` is passed to the `platform` which errors, but then the output of th command is handled by --json-output. What's the purpose of that argument, I mean, I see what it does, but I wonder what are some useful cases on those.

And the fact that we might be hiding it from the help doesn't make it stop working. I am hiding it when I disable flags because there it won't actually work even if used.

I think this is really something that eventually can be handled with more selective target of that global command on #4287.

In the meantime I say we just hide it when we are disabling flag parsing.